components/scream/src/physics/spa/spa_functions_impl.hpp

```diff
@@ -566,29 +589,42 @@ ::update_spa_data_from_file(
     // PS is defined only over columns
     ps_h(tgt_col) += PS_v_h(src_col)*src_wgt;
     // CCN3 and all AER variables have levels
-    for (int kk=0; kk<spa_horiz_interp.source_grid_nlevs; kk++) {
+    for (int kk=1; kk<spa_data.nlevs; kk++) {
       // Note, all variables we map to are packed, while all the data we just loaded as
       // input are in real N-D views.  So we need to back out the indices for the target
       // data.
       int pack = kk / Spack::n; 
       int kidx = kk % Spack::n;
-      ccn3_h(tgt_col,pack)[kidx] += CCN3_v_h(src_col,kk)*src_wgt;
+      // Note, we want to pad the actual source data such that
+      //   Y[0] = 0.0, note this is handled by the deep copy above
+      //   Y[nlevs-1] = Y[nlevs-2]
+      // and the data level index is kk-1 for kk = 1,nlevs-2
+      const int ksrc = kk < spa_data.nlevs-1 ? kk-1 : kk-2;
+      ccn3_h(tgt_col,pack)[kidx] += CCN3_v_h(src_col,ksrc)*src_wgt;
       for (int n=0; n<nswbands; n++) {
-        aer_g_sw_h(tgt_col,n,pack)[kidx]   += AER_G_SW_v_h(src_col,n,kk)*src_wgt;
-        aer_ssa_sw_h(tgt_col,n,pack)[kidx] += AER_SSA_SW_v_h(src_col,n,kk)*src_wgt;
-        aer_tau_sw_h(tgt_col,n,pack)[kidx] += AER_TAU_SW_v_h(src_col,n,kk)*src_wgt;
+        aer_g_sw_h(tgt_col,n,pack)[kidx]   += AER_G_SW_v_h(src_col,n,ksrc)*src_wgt;
+        aer_ssa_sw_h(tgt_col,n,pack)[kidx] += AER_SSA_SW_v_h(src_col,n,ksrc)*src_wgt;
+        aer_tau_sw_h(tgt_col,n,pack)[kidx] += AER_TAU_SW_v_h(src_col,n,ksrc)*src_wgt;
       }
       for (int n=0; n<nlwbands; n++) {
-        aer_tau_lw_h(tgt_col,n,pack)[kidx] += AER_TAU_LW_v_h(src_col,n,kk)*src_wgt;
+        aer_tau_lw_h(tgt_col,n,pack)[kidx] += AER_TAU_LW_v_h(src_col,n,ksrc)*src_wgt;
       }
     }
   }
-  for (int kk=0; kk<spa_horiz_interp.source_grid_nlevs; kk++) {
+  // We also need to pad the hyam and hybm views with
+  //   hya/b[0] = 0.0, note this is handled by deep copy above
+  //   hya/b[nlevs-1] = BIG number so always bigger than likely pmid for target
+  for (int kk=1; kk<spa_data.nlevs-1; kk++) {
     int pack = kk / Spack::n; 
     int kidx = kk % Spack::n;
-    hyam_h(pack)[kidx] = hyam_v_h(kk);
-    hybm_h(pack)[kidx] = hybm_v_h(kk);
+    hyam_h(pack)[kidx] = hyam_v_h(kk-1);
+    hybm_h(pack)[kidx] = hybm_v_h(kk-1);
   }
+  const int pack = (spa_data.nlevs-1) / Spack::n;
+  const int kidx = (spa_data.nlevs-1) % Spack::n;
+  hyam_h(pack)[kidx] = 1e5; 
```

Comment:
Since p_mid(end) will _never_ be 10x larger than p_src(end), hybm(end)=10 is probably way more than enough.

That said, 1e5 works.

Reply:
Actually, hold on. It's hybm that has to be large, and hyam=0, since the formula is
```
p_ref(k) = ps*hybm(k) + p0*hyam(k);
```

Reply:
I went with hyam because no matter what p0=10000, since its a reference surface pressure.  PS can be anything.  presumably it should never be small.  But P0 is a guarantee

Reply:
That being said, if switching to PS makes more sense I'm happy with that.

Reply:
Ah, yeah, you're right, it's the same.